File: ComputerScience/Algorithm/LeetCode/src/main/java/ch05_string/P02_Reverse_String.java
package ch05_string;

public class P02_Reverse_String {
  public void reverseString(char[] s) {
    int start = 0;
    int end = s.length - 1;

    while(start < end) {
      // 양 끝의 문자를 교환
      char temp = s[start];
      s[start] = s[end];
      s[end] = temp;

      // 포인터 이동
      start++;
      end--;
    }
  }
}

File: ComputerScience/Algorithm/LeetCode/src/main/kotlin/ch05_string/P02_Reverse_String.kt
package ch05_stringclass
class Solution_ {
    fun reverseString(s: CharArray): Unit {
        var start = 0
        var end = s.size - 1

        // 서로 중앙으로 나가다 겹치는 지점에 도달하면 종료
        while (start < end) {
            // also를 이용한 우아한 스왑
            s[start] = s[end].also { s[end] = s[start] }

            // start는 한 칸 뒤로, end는 한 칸 앞으로 이동
            start++
            end--
        }
    }
}